packit/command_handler.py

```diff
@@ -125,17 +125,27 @@ def sandcastle(self):
         """initialize Sandcastle lazily"""
         if self._sandcastle is None:
             # we import here so that packit does not depend on sandcastle (and thus python-kube)
-            from sandcastle.api import Sandcastle, MappedDir
+            from sandcastle.api import Sandcastle, MappedDir, VolumeSpec
 
             self._mapped_dir = MappedDir(
                 local_dir=self.local_project.working_dir,
                 path=self.config.command_handler_work_dir,
                 with_interim_pvc=True,
             )
+            pvc_volume_specs = [
+                VolumeSpec(
+                    path=vol_spec_dict.get("path"),
+                    volume_name=vol_spec_dict.get("volume_name"),
+                    pvc=vol_spec_dict.get("pvc"),
```

Comment:
I just checked sandcastle's code and it seems these 2 options are exclusive: so either you have a PVC which means that kube will select a volume from its pool or you pick an existing volume and assign it to the pod - no claim hustling.

There is no test case for this scenario (because we never used it like that). I'd suggest adding one to make sure it does what you expect it to do - probably even mimic the repo cache behaviour.

Reply:
Kubernetes is magic, it works for me out of the box: the volume is created on-demand without doing anything: https://github.com/packit/sandcastle/pull/135

Now that I look at sandcastle code closer, it's actually valid to specify volume name and the claim name, because volume name is used for referencing: sandcastle would need to generate it.